ibis/pandas/client.py

```diff
@@ -41,10 +43,20 @@ def pandas_dtypes_to_ibis_schema(df):
                 'Column names must be strings to use the pandas backend'
             )
 
-        if dtype == np.object_:
-            ibis_type = _INFERRED_DTYPE_TO_IBIS_TYPE[
-                infer_dtype(df[column_name].dropna())
-            ]
+        if column_name in schema:
+            ibis_type = dt.validate_type(schema[column_name])
+        elif dtype == np.object_:
+            inferred_dtype = infer_dtype(df[column_name].dropna())
```

Comment:
Yikes. I guess we should make a NaN-friendly type inference function someplace (seems like an oversight in infer_dtype originally)